### ISSUE TITLE
FR: Support ntfy formatting for webhooks

### ISSUE THREAD

**Issue by Zocker1999NET** [tailscale/tailscale#9340]
### What are you trying to do?

I want to have notifications on changes to my tailnet using webhooks. If possible, I want them to be fancy. My phone is already connected to a self-hosted [ntfy](https://ntfy.sh/) instance due to UnifiedPush. I’m not using Slack, Discord or Mattermost (and would rather avoid using them).

### How should we solve this?

Implement the message format of ntfy as optional selection (similar to Discord, Slack or Mattermost support). ntfy defines [documentation](https://docs.ntfy.sh/publish/#publishing) It should be continued to support custom webhook URLs for self-hosted ntfy instances and custom topic names.

### What is the impact of not solving this?

Not so fancy notifications. My current workaround is to just send the normal JSON webhook, as I do not assume to have many notifications. If I receive one, I get a notification, but need to decipher the JSON manually.

### Anything else?

This feature is not really required, but it would be fancy & I hope implementing that should be easy as it should behave similar like other services do. The only caveat might be that the custom format uses HTTP headers when POSTing to the topic URLs directly (see [docs](https://docs.ntfy.sh/publish/#publishing)). If you want to POST a JSON document, POSTing to the base URL is possible (see [docs](https://docs.ntfy.sh/publish/#publish-as-json)), however this would require asking the user for an additional topic name OR you separate topic name from the base URL based on the user input (e.g. `https://ntfy.sh/topic_name`).

ntfy is also a nice, easy & energy-saving solution to receive custom notifications on Android, which is why I use it.

Thanks for your work in Tailscale & thanks for considering this feature!

**Comment by purple-emily** [comment]
This would be great!

**Comment by lazee486** [comment]
ntfy is a publish/subscribe alert system...so i can subscribe from multiple devices where i want those alerts...for business use a team could get alerts for certain things...  plus it can be self hosted completely on tailscale for secure alerts from my whole homelab...  maybe look at [apprise](https://github.com/caronc/apprise)  for a more prolific notification library so you dont need to support each individually...

**Comment by antreos** [comment]
I encountered the same issue and came across a solution. After experimenting and receiving quick assistance from wunter8 on the ntfy Discord, I got this workaround to function:

Webhook URL: `https://ntfy.sh/topic_name?tpl=1&m={{.content}}&md=yes`
Destination: `Discord`

Here's how it looks:
![demo-notification](https://github.com/user-attachments/assets/3a3ef32f-3c8f-40e9-a484-ed555949c133)

While it's just a workaround and ideally Tailscale should support this feature natively, it's better than nothing.

**Comment by Zydepoint** [comment]
> I encountered the same issue and came across a solution. After experimenting and receiving quick assistance from wunter8 on the ntfy Discord, I got this workaround to function:
> 
> Webhook URL: `https://ntfy.sh/topic_name?tpl=1&m={{.content}}&md=yes` Destination: `Discord`
> 
> Here's how it looks: ![demo-notification-v2](https://github.com/user-attachments/assets/6c7bbf40-5ff8-4159-a700-6a9a24e1c15e)
> 
> While it's just a workaround and ideally Tailscale should support this feature natively, it's better than nothing.

Just tried this and it works great actually! Of course native support would be the best, but this does the job.

**Comment by Frogerman1234** [comment]
I agree this would be great if natively supported. They could even make their own template that we can download and put on your NTFY instance to make it pretty.

**Comment by Klaus-Kleber** [comment]
+1 
Having the same issue
